inst/lua/counter.lua

```diff
@@ -0,0 +1,18 @@
+local utils = require 'pandoc.utils'
+local counter = 0
+local a = {}
+
+function Code (code)
+  if string.find(utils.stringify(code), '@@') then
+    ref_id = string.gsub(utils.stringify(code), "@@(.)", "\\%1")
+    if(a[ref_id] == nil) then
+      counter = counter + 1
+      ref_id =  string.gsub(utils.stringify(code), "@@(.)", "\\%1")
+      a[ref_id] = counter
+
```

Comment:
Super cool! in the dynamic referencer, we have an argument for "type" as well so numbering of specs does not impact the numberin not sure how we would denote that with a @@TYPE:reference? more similar to the latex \ref(fig:referenceid)

Reply:
so like @@spec:specification-1, @@test_case:case1, @@test_code:test145

Reply:
I think that should be possible. But still wondering if we are okay that the `@@type:ref` has to be marked inline code 🤔 Does that seem strange?

Reply:
Oh, missed that! hmm, not sure. Thats so it can be id'd with code? could we set it up to be a more generic "find cases of "@@(([[:alnum:]]|[_]|[-])*)):(([[:alnum:]]|[_]|[-])*))" in the doc and then parse/count?

Looks like code is being used to delim the words essentially?

Reply:
> For some strange reason when trying to map @@ to str elements (which would let us avoid the backticks) pandoc/lua is having issues with the @@ specifically - I suspect there is another filter conflicting 

but I am not very experienced at lua coding, so could be missing something (🤞 obvious). see commit fa15c08 which shows how the logic is being scraped different.